### PR TITLE
k8s single cert

### LINKED ISSUE #1608: kubernetes should use the same certs as everything else
No reason not to, and it'll reduce attack surface, complexity, and code.  

Looks like the certs were valid for the following SAN (from `roles/kubernetes/files/make-ca-cert.sh`):
```
dns_domain="${DNS_DOMAIN:="cluster.local"}"
sans="IP:${service_ip},IP:127.0.0.1,DNS:localhost,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.${dns_domain}"
```
That script is called with the following environment:
```yml
  environment:
    MASTERS: "kubernetes.service.consul"
    DNS_DOMAIN: "{{ dns_domain }}"
    SERVICE_CLUSTER_IP_RANGE: "{{ kube_service_addresses }}"
    CERT_DIR: "{{ playbook_dir }}/{{ cert_syncdir.path }}/certs"
```
so the SAN will really include:
```
DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.service,DNS:kubernetes.default.svc.cluster.local,DNS:kubernetes.service.consul,DNS:kubernetes.default.src.{{ cluster_name or dns_domain }}
```
I don't think we need to have our certs valid for all those names though. Can anyone think of a reason to? 

- Ansible version (`ansible --version`): 1.9.4
- Python version (`python --version`): 2.7.11
- Git commit hash or branch: master
- Cloud Environment: all
- Terraform version (`terraform version`): 0.7.0

### PR DESCRIPTION
**Pull request by siddharthist**
I think this maybe finally works! 

check the commit messages for a more detailed outline of what happened in each commit.

fixes #1608 

- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

### CODE REVIEW

**Comment by siddharthist** [comment]
Looks like we need to add Python's `netaddr` module to our `requirements.txt`.

**Comment by siddharthist** [comment]
@sehqlr This is also caught by the same bug where the CI build doesn't actually get the requirements from `requirements.txt`:
 - https://travis-ci.org/CiscoCloud/mantl/jobs/146636961#L299
 - https://travis-ci.org/CiscoCloud/mantl/jobs/146636961#L1254

Is there something we can do about this in the short term? I'm opening an issue to track progress.

File: roles/kubernetes-master/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 ---
+# Consul must be available when kubelet stores the manifests in the k/v.
+- name: wait for leader
+  command: /usr/local/bin/consul-wait-for-leader.sh
+  tags:
+    - consul
```
Comment (ryane) [inline roles/kubernetes-master/tasks/main.yml]: should this task have the kubernetes tag?

**Comment by ryane** [comment]
just noticed one tiny potential issue with an ansible tag but otherwise this worked flawlessly for me on aws. it does look like the build failures are related to the ci environment.

File: roles/kubernetes-master/tasks/main.yml (hunk 1 above)
Comment (siddharthist) [inline roles/kubernetes-master/tasks/main.yml]: Yep!

Comment (siddharthist) [inline roles/kubernetes-master/tasks/main.yml]: This is done in the latest commit

**Comment by siddharthist** [comment]
Since the CI build was fixed in #1758, this is good to go!

**Comment by Dacode45** [comment]
Hey guys, I'm not really sure how, but pods no longer seem to be getting a working certificate. 
All pods should get certs from `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` I've been trying to deploy an ingress controller with 
```
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: traefik-ingress-controller
  labels:
    k8s-app: traefik-ingress-lb
spec:
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb
        name: traefik-ingress-lb
    spec:
      terminationGracePeriodSeconds: 60
      hostNetwork: true
      nodeSelector:
        role: edge
      volumes:
        - name: config-volume
          configMap:
            name: traefik-ingress-controller-config
      containers:
        - image: traefik
          name: traefik-ingress-lb
          imagePullPolicy: Always
          ports:
            - containerPort: 80
              hostPort: 80
            - containerPort: 443
              hostPort: 443
            - containerPort: 8080
          volumeMounts:
            - name: config-volume
              mountPath: /etc/traefik
          args:
            - -c
            - /etc/traefik/traefik.toml
            - --web
            - --logLevel=DEBUG
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: traefik-ingress-controller-config
data:
  traefik.toml: |-
    port = ":80"
    graceTimeOut = 10
    logLevel = "DEBUG"

    [web]
    address = ":8080"
    CertFile = "/etc/pki/mantl/cert"
    KeyFile = "/etc/pki/mantl/key"

    [marathon]
    endpoint = "http://marathon.service.consul:18080"
    domain = "marathon.localhost"
    watch = true
    networkInterface = "eth0"
    ExposedByDefault = true

    [kubernetes]
```
Which worked yesterday. Now when I run it, traefik complains 
```
level=error msg="Error watching kubernetes events: failed to create watch: failed to do
 version request: GET \"https://10.254.0.1:443/apis/extensions/v1beta1/ingresses\" : http error 401 GET \"https://1
0.254.0.1:443/apis/extensions/v1beta1/ingresses\": \"Unauthorized\\n\"" 
```

**Comment by siddharthist** [comment]
@Dacode45 Can you open a separate issue, please? thanks

**Pull request merged**